web_src/js/features/repo-issue.js

```diff
@@ -226,18 +226,43 @@ export function initRepoIssueCodeCommentCancel() {
 export function initRepoIssueStatusButton() {
   // Change status
   const $statusButton = $('#status-button');
+  // init value
+  $('#state').val($statusButton.data('value'));
+  $('#status').val($statusButton.data('status-val'));
   $('#comment-form textarea').on('keyup', function () {
     const easyMDE = getAttachedEasyMDE(this);
     const value = easyMDE?.value() || $(this).val();
-    $statusButton.text($statusButton.data(value.length === 0 ? 'status' : 'status-and-comment'));
+    // find active item of dropdown menu
+    const $active = $('#status-dropdown').find('> .active');
+    $statusButton.text($active.data(value.length === 0 ? 'status' : 'status-and-comment'));
   });
   $statusButton.on('click', (e) => {
     e.preventDefault();
-    $('#status').val($statusButton.data('status-val'));
     $('#comment-form').trigger('submit');
   });
 }
 
+export function initRepoIssueStatusDropdown() {
+  const $statusDropdown = $('#status-dropdown');
+  const $statusMenu = $statusDropdown.find('> .menu');
+  $statusMenu.find('> .item').each((_, item) => {
+    $(item).on('click', (e) => {
```

Comment:
I guess a lot of code could be simplified by Dropdown's builtin mechisam

The dropdown could be used as a `<select>`-like component, it can have an internal `input`, it can remember it's internal value and update the UI by the internal value.